docs/adr/0003-use-of-devcontainer-directory.md

```diff
@@ -0,0 +1,55 @@
+# 3. Use of the .devcontainer directory in the research-template repo
```

Comment:
I've said this for the other two ADRs, so for completeness I'll say it for this one: Nygard suggests the title should be a noun phrase. If it's a noun phrase, then it should read right with "the", "a", or "an" in front of it. For example: "the research-template repository's `.devcontainer` directory".